src/tools/plan-generator.ts

```diff
@@ -67,11 +79,20 @@ export class PlanGenerator {
     ]);
   }
 
+  /** Aggregate mapping of schema hashes and schema properties from particle connections */
+  async collectParticleConnectionSpecs(particle: Particle): Promise<void> {
+    for (const connection of particle.spec.connections) {
+      const specName = `${particle.spec.name}_${upperFirst(connection.name)}_Spec`;
```

Comment:
Might be an opportunity for another codegen helper function: `joinWithUnderscores(particle.spec.name, connection.name, 'Spec')` (could do correct uppercasing logic for you too).

I'm really liking these codegen helper functions btw!

Reply:
Better yet: let's make it `joinWith`, and take a user-specified delimiter.

Reply:
Thanks!

Reply:
I think this is just the `join` function on arrays.

Reply:
I'm glad you like them!